src/EvolvableLinkTrait.php

```diff
@@ -19,13 +19,13 @@ trait EvolvableLinkTrait
      *
      * @return EvolvableLinkInterface
      */
-    public function withHref($href)
+    public function withHref(\Stringable|string $href): static
```

Comment:
Is there a significance to this being the only string param that's also Stringable?  Is it a PSR-7 related assumption?

Reply:
For this PR, the significance is that's what the updated interface spec itself uses so I have to follow suit here.

The reason for that decision was that a Url is a complex string, so having an object version of it that's easily modifiable and casts down to a string (whether PSR-7's or otherwise) is a reasonable use case.  Contrast with a simple key, which is an opaque unstructured string so expecting it to be a for-reals string is reasonable.  That's a judgement call, but it was already made when updating PSR-13 itself.  This is just following suit.